### PR TITLE
WeekDaysNames component

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Move WeekDaysNames component to commons dir and use it in other components

### CODE REVIEW

File: src/agenda/style.ts [hunk 1]
```diff
@@ -34,13 +34,13 @@ export default function styleConstructor(theme: Theme = {}) {
       alignItems: 'center',
       backgroundColor: appStyle.calendarBackground
     },
-    weekday: {
+    dayHeader: {
```
Comment (ethanshar) [inline src/agenda/style.ts:37]: Shouldn't we call it `daysHeader`? 
btw here and in other places.. 

```suggestion
    daysHeader: {
```

File: src/calendar/header/index.tsx [hunk 2]
```diff
@@ -235,11 +235,15 @@ const CalendarHeader = forwardRef((props: CalendarHeaderProps, ref) => {
...
+    return showWeekNumbers && <View style={style.current.dayHeader}/>;
+  };
+
   const renderDayNames = () => {
     if (!hideDayNames) {
       return (
         <View style={style.current.week} testID={testID ? `${HEADER_DAY_NAMES}-${testID}` : HEADER_DAY_NAMES}>
-          {showWeekNumbers && <Text allowFontScaling={false} style={style.current.dayHeader}></Text>}
+          {renderWeekNumbersSpace()}
           {renderWeekDays}
```
Comment (ethanshar) [inline src/calendar/header/index.tsx:247]: Why didn't you migrate here to WeekDaysNames component as well?

File: src/calendar/header/index.tsx [hunk 3]
```diff
@@ -235,11 +235,15 @@ const CalendarHeader = forwardRef((props: CalendarHeaderProps, ref) => {
...
+  const renderWeekNumbersSpace = () => {
+    return showWeekNumbers && <View style={style.current.dayHeader}/>;
+  };
+
   const renderDayNames = () => {
     if (!hideDayNames) {
       return (
         <View style={style.current.week} testID={testID ? `${HEADER_DAY_NAMES}-${testID}` : HEADER_DAY_NAMES}>
-          {showWeekNumbers && <Text allowFontScaling={false} style={style.current.dayHeader}></Text>}
+          {renderWeekNumbersSpace()}
```
Comment (ethanshar) [inline src/calendar/header/index.tsx:246]: What's the purpose of the Space render here?

File: src/agenda/style.ts (hunk 1 above)
Comment (Inbal-Tish) [reply src/agenda/style.ts:37]: It's a style that repeats on other components for the same part. I would like to keep it this way and not change on all files. The aim is to make it the default style and delete it eventually.

File: src/calendar/header/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/header/index.tsx:247]: It is an overlay for the vertical calendar. When you have the week numbers, the calendar offset a day and this means adding the same offset to the weekdays names

File: src/calendar/header/index.tsx (hunk 3 above)
Comment (Inbal-Tish) [reply src/calendar/header/index.tsx:246]: It is an overlay for the vertical calendar. When you have the week numbers, the calendar offset a day and this means adding the same offset to the weekdays names

File: src/calendar/header/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/header/index.tsx:247]: This header has more functionality, mainly the 'dayTextAtIndex', passing this functionality to WeekDaysNames component will complicate the code as it is adding styles to the header/style StyleSheet

Comment (ethanshar) [reply src/calendar/header/index.tsx:247]: IMO it makes more sense to make the WeekDaysNames more generic and maybe a little more complex but support all use cases accordingly. 
Anyway, it's up to you, we can also migrate this use case later on..

**Review by ethanshar — APPROVED** [review]

File: src/calendar/header/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/header/index.tsx:247]: That was my first thought, and I tried it, but then it became very complex and hard to follow that I decided to leave it for now. We can revisit it in the future.
